sklearn/neighbors/_base.py

```diff
@@ -391,10 +391,10 @@ class NeighborsBase(MultiOutputMixin, BaseEstimator, metaclass=ABCMeta):
 
     _parameter_constraints = {
         "n_neighbors": [Interval(Integral, 1, None, closed="left"), None],
-        "radius": [Interval(Integral, 1, None, closed="left"), None],
+        "radius": [Interval(Real, 0, None, closed="both"), None],
         "algorithm": [StrOptions({"auto", "ball_tree", "kd_tree", "brute"})],
         "leaf_size": [Interval(Integral, 1, None, closed="left")],
-        "p": [Interval(Integral, 1, None, closed="left")],
+        "p": [Interval(Real, 1, None, closed="both")],
```

Comment:
Currently we allow setting `p=None` when `metric not in {"wminkowski", "minkowski"}`
```suggestion
        "p": [Interval(Real, 1, None, closed="both"), None],
```